packages/swap/src/routes/v2/quote.ts

```diff
@@ -234,29 +234,37 @@ export const generateQuotes = async ({
   const boostedQuote = getFulfilledResult(boostedQuoteResult, null);
   const dcaBoostedQuote = getFulfilledResult(dcaBoostedQuoteResult, null) as DCABoostQuote | null;
 
-  if (dcaQuoteParams && dcaQuote) {
-    // The received quotes are for a single chunk, we need to extrapolate them to the full amount
-    adjustDcaQuote({ dcaQuoteParams, dcaQuote, originalDepositAmount: amount });
-    if (dcaBoostedQuote) {
-      adjustDcaQuote({ dcaQuoteParams, dcaQuote: dcaBoostedQuote, originalDepositAmount: amount });
-    }
-
-    // Check liquidity for DCA
+  const validateLiquidity = async (egressAmount: string, intermediateAmount?: string) => {
     if (srcAsset === 'Usdc' || destAsset === 'Usdc') {
       const totalLiquidity = await getTotalLiquidity(srcAsset, destAsset);
-      if (totalLiquidity < BigInt(dcaQuote.egressAmount)) {
+      if (totalLiquidity < BigInt(egressAmount)) {
         throw ServiceError.badRequest(`Insufficient liquidity for the requested amount`);
       }
     } else {
       const totalLiquidityLeg1 = await getTotalLiquidity(srcAsset, 'Usdc');
       const totalLiquidityLeg2 = await getTotalLiquidity('Usdc', destAsset);
       if (
-        totalLiquidityLeg1 < BigInt(dcaQuote.intermediateAmount!) ||
-        totalLiquidityLeg2 < BigInt(dcaQuote.egressAmount)
+        totalLiquidityLeg1 < BigInt(intermediateAmount!) ||
+        totalLiquidityLeg2 < BigInt(egressAmount)
       ) {
         throw ServiceError.badRequest(`Insufficient liquidity for the requested amount`);
       }
     }
+  };
+
+  if (dcaQuoteParams && dcaQuote) {
+    // The received quotes are for a single chunk, we need to extrapolate them to the full amount
+    adjustDcaQuote({ dcaQuoteParams, dcaQuote, originalDepositAmount: amount });
+    if (dcaBoostedQuote) {
+      adjustDcaQuote({ dcaQuoteParams, dcaQuote: dcaBoostedQuote, originalDepositAmount: amount });
+    }
+
+    // Check liquidity for DCA
+    await validateLiquidity(dcaQuote.egressAmount, dcaQuote.intermediateAmount);
+  } else if (quote) {
+    await validateLiquidity(quote.egressAmount, quote.intermediateAmount);
+  } else {
+    throw ServiceError.internalError('could not generate quotes');
```

Comment:
i think we should always do this for all of the quotes, no?

```suggestion
  } 
  
  if (quote) {
    await validateLiquidity(quote.egressAmount, quote.intermediateAmount);
  }
```